lib/sport_ngin_aws_auditor/scripts/audit.rb

```diff
@@ -46,51 +47,61 @@ def self.gather_data(class_type, tag_name)
 
         if options[:instances]
           instances = klass.get_instances(tag_name)
+          retired_tags = klass.get_retired_tags(instances)
           instances_with_tag = klass.filter_instances_with_tags(instances)
           instances_without_tag = klass.filter_instance_without_tags(instances)
           instance_hash = klass.instance_count_hash(instances_without_tag)
           klass.add_instances_with_tag_to_hash(instances_with_tag, instance_hash)
         elsif options[:reserved]
           instance_hash = klass.instance_count_hash(klass.get_reserved_instances)
         else
-          instance_hash = klass.compare(tag_name)
-          retired_reserved_instances = klass.get_recent_retired_reserved_instances
+          instance_hash, retired_tags = klass.compare(tag_name)
+          retired_ris = klass.get_recent_retired_reserved_instances
         end
 
         compared_array = []
         instance_hash.each do |key, value|
           compared_array.push(Instance.new(key, value))
         end
 
-        [compared_array, retired_reserved_instances]
+        [compared_array, retired_tags, retired_ris]
```

Comment:
Returning arrays of complex objects technically works fine, but the complexity makes for poor readability and brittle code.  Consider refactoring and encapsulating in a class.